components/windows/swap-widget.tsx

```diff
@@ -29,6 +31,32 @@ export const SwapWidget = () => {
     formatUnits(ethBalance.data ? ethBalance.data.value : BigInt(0), 18)
   ).toPrecision(4);
 
+  const ethPrice = useEthPrice();
+
+  useEffect(() => {
+    if (ethPrice) {
+      if (isSwapped) {
+        const tokenAmountValue = tokenAmount.length ? Number(tokenAmount) : 0;
+        const usdValue = tokenAmountValue * psyDAOTokenPrice;
+        const ethValue = usdValue / ethPrice;
+        if (!isNaN(ethValue)) {
+          setEthAmount(ethValue.toFixed(2));
+        } else {
+          setEthAmount("0.00");
+        }
+      } else {
+        const ethAmountValue = ethAmount.length ? Number(ethAmount) : 0;
+        const usdValue = ethAmountValue * ethPrice;
+        const tokens = usdValue / psyDAOTokenPrice;
+        if (!isNaN(tokens)) {
+          setTokenAmount(tokens.toFixed(2));
+        } else {
+          setTokenAmount("0.00");
+        }
+      }
+    }
+  }, [ethAmount, ethPrice, isSwapped, tokenAmount]);
```

Comment:
Can we move this into a function that will return a value.
Both of codes look similar L39-L46 and L48-L55.
Lets refactor it a bit :)